src/main/scala/ch/epfl/bluebrain/nexus/kg/async/DistributedCache.scala

```diff
@@ -273,27 +292,57 @@ object Projects {
         case _                                       => Future.successful(false)
       }
 
-    private def update(ref: ProjectRef, proj: Project): Future[Boolean] = {
+    private def updateProject(ref: ProjectRef, proj: Project): Future[Boolean] = {
       val empty  = LWWRegister(RevisionedValue[Option[Project]](0L, None))
       val value  = RevisionedValue[Option[Project]](proj.rev, Some(proj))
       val update = Update(projectKey(ref), empty, WriteMajority(tm.duration))(_.withValue(value))
       (replicator ? update).flatMap(handleBooleanUpdate("Timed out while waiting for add project quorum response"))
     }
 
-    private def update(ref: ProjectRef, accountRef: AccountRef, proj: Project): Future[Boolean] =
-      update(ref, proj).withFilter(_ == true).flatMap { _ =>
-        accountSegment(accountRef).flatMap {
-          case Some(accountL) =>
-            val empty = LWWRegister(RevisionedValue[Option[ProjectRef]](0L, None))
-            val value = RevisionedValue[Option[ProjectRef]](proj.rev, Some(ref))
-            val update = Update(projectSegmentKey(ProjectLabel(accountL, proj.label)),
-                                empty,
-                                WriteMajority(tm.duration))(_.withValue(value))
-            (replicator ? update).flatMap(
-              handleBooleanUpdate("Timed out while waiting for add project quorum response"))
-          case _ => Future.successful(false)
+    private def updateProjectLabel(ref: ProjectRef,
+                                   accountRef: AccountRef,
+                                   proj: Project,
+                                   instant: Instant): Future[Boolean] = {
+      updateProject(ref, proj).withFilter(identity).flatMap { _ =>
+        addProjectToAccount(ref, accountRef, instant).withFilter(identity).flatMap { _ =>
+          accountSegment(accountRef).flatMap {
+            case Some(accountLabel) =>
+              val empty = LWWRegister(RevisionedValue[Option[ProjectRef]](0L, None))
+              val value = RevisionedValue[Option[ProjectRef]](proj.rev, Some(ref))
+              val update = Update(projectSegmentKey(ProjectLabel(accountLabel, proj.label)),
+                                  empty,
+                                  WriteMajority(tm.duration))(_.withValue(value))
+              (replicator ? update).flatMap(
+                handleBooleanUpdate("Timed out while waiting for add project quorum response"))
+            case _ => Future.successful(false)
+          }
+        }
+      }
+    }
+
+    private def addProjectToAccount(ref: ProjectRef, accountRef: AccountRef, instant: Instant): Future[Boolean] = {
+      projects(accountRef).flatMap { projects =>
+        if (projects.contains(ref)) Future.successful(true)
```

Comment:
When it already exists, do we signal it as true ? I think [in some places](https://github.com/hygt/nexus-kg/blob/9a03abf7d8bb14f9395758390b714abd3d0bae95/src/main/scala/ch/epfl/bluebrain/nexus/kg/async/DistributedCache.scala#L412) we had the opposite logic (return false if it already exists). Seems that we only return true when actually something gets added/deleted/updated and not when the thing is already existing